Netflix/consoleme
# fix authentication cookie issue

This is the proof of concept of how to handle 401 status so that users can be informed and provide a chance to reauthenticate if the session is expired.

Followings are the changes done:
- added a state called isSessionExpired to track session expiration status.
- moved sendRequestCommon to be part of AuthProvider hook so that if 401 status is returned from the backend via sendRequestCommon then it can set the states in AuthProvider which sits on top of any other components in ConsoleMe to control things globally.
- Searched places where sendRequestCommon is being used and replaced with this logic including components that are not yet React Hooks.
- Components in blocks directory and some of helper util.js are not updated yet which requires more work. I will do this once this proof of concept is valid.

How to test this:
- First visit Self Service Page then it will pop up a modal for reauthentication.
- I have updated the backend endpoint /api/v2/self_service_config to return 401 via AJAX and sendRequestCommon will set isSessionExpired to true so that a Modal will be returned to allow people to reauthenticate
- Once you click the button in the modal, we want to redirect the user to the IDP for login once more. The redirect URL should be configurable and returned from the backend so that it can bet set to the state and the modal can use it.
